internal/pkg/client/cache/library_test.go

```diff
@@ -35,16 +40,57 @@ func TestLibrary(t *testing.T) {
 	}
 }
 
-// Will come back to after functionality exists
-// func TestLibraryImageExists(t *testing.T) {
-// 	tests := []struct {
-// 		name     string
-// 		env      string
-// 		expected string
-// 		file     []struct {
-// 			create bool
-// 			sum    string
-// 			name   string
-// 		}
-// 	}{}
-// }
+func TestLibraryImage(t *testing.T) {
+	LibraryImage("", "")
+}
+
+func TestLibraryImageExists(t *testing.T) {
+	// Invalid cases
+	_, err := LibraryImageExists("", "")
+	if err == nil {
+		t.Fatalf("LibraryImageExists() returned true for invalid data:  %s\n", err)
+	}
+
+	// Pull an image so we know for sure that it is in the cache
+	sexec, err := exec.LookPath("singularity")
```

Comment:
this requires singularity to be installed for the test to be able to run. I think we should have some way of skipping tests like this. It should at least check that `testing.Short()` is not set